File: snake-ai/scripts/tryModel.ts
import GaModel from "../GaModel";
import { setting } from "./setting";

async function main() {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const print: any[] = [];

  const gaModel = new GaModel(setting.gaPlayerConfig, 10);
  for (let i = 0; i < setting.evolveTimes; i++) {
    const result = await gaModel.evolve();
    print.push({
      generation: i,
      bestFitness: result.bestIndividual.fitness,
      bestLength: result.bestIndividual.snakeLength,
      meanFitness: result.overallStats.fitness.mean,
      meanLength: result.overallStats.snakeLength.mean,
      timeSpent: result.timeSpent,
    });
    console.table(print.slice(-50));
  }
}

main();

File: kotlin-root/gasnake/src/main/kotlin/snake/ai/ga/trymodel/TryModel.kt
package com.example.snake.ai.ga.trymodel

import com.example.snake.ai.ga.GaModel
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.runBlocking
import snake.ai.ga.trymodel.setting

fun main() = runBlocking {
    val printList = mutableListOf<Map<String, Any>>()

    val gaModel = GaModel(setting.gaPlayerConfig, 20)

    launch(Dispatchers.Default) {
        for (i in 0 until setting.evolveTimes) {
            val result = gaModel.evolve()
            printList.add(
                mapOf(
                    "generation" to i,
                    "bestFitness" to result.bestIndividual.fitness,
                    "bestLength" to result.bestIndividual.snakeLength,
                    "meanFitness" to result.overallStats.fitness.mean,
                    "meanLength" to result.overallStats.snakeLength.mean,
                    "timeSpent" to result.timeSpent
                )
            )
            println(printList.last())
        }
    }

    Unit
}